lib/fluent/parser.rb

```diff
@@ -345,6 +352,16 @@ def convert_field_type!(record)
           end
         }
       end
+
+      def convert_value_to_nil(value)
+        if @null_empty_string
+          value = (value == '') ? nil : value
+        end
+        if @null_value_pattern
+          value = (value =~ @null_value_pattern) ? nil : value
```

Comment:
I usually does String#scrub in my plugin as https://github.com/fluent/fluentd/blob/master/lib/fluent/plugin/filter_grep.rb#L73-L83. 
I am not sure for the case of fluentd itself. What do you think? > @repeatedly